### PR TITLE
gamehub: init at 0.16.3-2

### LINKED ISSUE #58189: package request: gamehub
## Issue description

repo: https://github.com/tkashkin/gamehub

originally mentioned at https://github.com/NixOS/nixpkgs/pull/42075#issuecomment-460632423.

### PR DESCRIPTION
**Pull request by pasqui23**
###### Motivation for this change
Fixes https://github.com/NixOS/nixpkgs/issues/58189

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by pasqui23** [comment]
Wait as soon as I try to open the screen with all the aviable games it goes segmentation fault ☹

**Comment by pasqui23** [comment]
It also freeze if I try to setup itch account even through I have it installed

**Comment by bbigras** [comment]
Is the https://tkashkin.tk/projects/gamehub/ URL good? It's "blocked" by ublock origin on my side.

**Comment by pasqui23** [comment]
Here is the core dump:
https://e1.pcloud.link/publink/show?code=XZjCbpZjM2kWKiP6MFWvJI3WD6DBBtt9dw7

File: pkgs/games/gamehub/default.nix [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  };
+
+  nativeBuildInputs = [
+    meson
+    ninja
+    vala
+    pkg-config
+    desktop-file-utils
+    wrapGAppsHook
+
```
Comment (AndersonTorres) [inline pkgs/games/gamehub/default.nix:40]: whitespace

```suggestion
```

File: pkgs/games/gamehub/default.nix [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  ];
+  propagatedBuildInputs = [
```
Comment (AndersonTorres) [inline pkgs/games/gamehub/default.nix:42]: lacks whitespace
```suggestion

  propagatedBuildInputs = [
```

File: pkgs/games/gamehub/default.nix [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    sqlite
+    webkitgtk
+    libmanette
+    libXtst
+    glib-networking
+  ];
+
+  meta = with lib; {
+    description = "All your games in one place";
+    homepage = "https://tkashkin.github.io/projects/gamehub";
```
Comment (AndersonTorres) [inline pkgs/games/gamehub/default.nix:57]:
```suggestion
    homepage = "https://tkashkin.github.io/projects/gamehub";
    description = "All your games in one place";
    longDescription = ''
      GameHub is a unified library for all your games. It allows you to store 
      your games from different platforms into one program to make it easier 
      for you to manage your games.
    '';
```

File: pkgs/games/gamehub/default.nix [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    meson
+    ninja
+    vala
+    pkg-config
+    desktop-file-utils
+    wrapGAppsHook
+
+  ];
+  propagatedBuildInputs = [
+    glib
```
Comment (AndersonTorres) [inline pkgs/games/gamehub/default.nix:43]: I suggest to put this is alphabetical ordering.

File: pkgs/games/gamehub/default.nix [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+  src = fetchFromGitHub {
+    owner = "tkashkin";
+    repo = pname;
+    rev = "${version}-master";
+    sha256 = "sha256-vp2AaTxrh37uUEwWeT7qlrWcxJqrrj4VOr8IKRrXHSI=";
+  };
+
+  nativeBuildInputs = [
+    meson
```
Comment (AndersonTorres) [inline pkgs/games/gamehub/default.nix:34]: Alphabetical

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Comment by pasqui23** [comment]
This is ready, my problems are all related to broken steam

**Comment by gador** [comment]
please address the failing tests

**Review by gador — CHANGES_REQUESTED** [review]
Hi, thanks for your PR!
A couple of points:
- please address the failing tests
- please address the requested changes from @AndersonTorres 
- please fill out the PR template and check the points

File: pkgs/games/gamehub/default.nix [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+    libsoup
+    libXtst
+    sqlite
+    webkitgtk
+  ];
+
+  meta = with lib; {
+    description = "All your games in one place";
+    homepage = "https://tkashkin.github.io/projects/gamehub";
+    description = "All your games in one place";
```
Comment (SuperSandro2000) [inline pkgs/games/gamehub/default.nix:58]:
```suggestion
```

File: pkgs/games/gamehub/default.nix [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+
+  meta = with lib; {
+    description = "All your games in one place";
+    homepage = "https://tkashkin.github.io/projects/gamehub";
+    longDescription = ''
+      GameHub is a unified library for all your games. It allows you to store
+      your games from different platforms into one program to make it easier
+      for you to manage your games.
+    '';
+    mantainers = with maintainers; [ pasqui23 ];
```
Comment (SuperSandro2000) [inline pkgs/games/gamehub/default.nix:63]:
```suggestion
    maintainers = with maintainers; [ pasqui23 ];
```

File: pkgs/top-level/all-packages.nix [hunk 8]
```diff
@@ -21638,6 +21638,8 @@ with pkgs;
   };
 
   fusionInventory = callPackage ../servers/monitoring/fusion-inventory { };
+
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:21641]:
```suggestion

```

**Comment by rapenne-s** [comment]
Result of `nixpkgs-review pr 158044` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>gamehub</li>
  </ul>
</details>

**Comment by rapenne-s** [comment]
While the program runs fine, it doesn't keep the credentials for the various services (dunno if voluntary or an upstream bug). I tried to install a game from GOG but it fails, hereis the error for a Linux game.

```
Verifying archive integrity... All good.
Uncompressing Darwinia (GOG.com)  100%  
Collecting info for this system...
Operating system: linux
CPU Arch: x86_64
trying mojosetup in bin/linux/x86_64
USING 
./startmojo.sh: ligne 55: /tmp/selfgz304947047/bin/linux/x86_64/mojosetup: No such file or directory
Error: Couldn't run mojosetup
/home/jeux/Games/_Collection/GOG/Darwinia - 10000th Anniversary Edition/darwinia_2_2_0_169_g706f93f4_gnu_linux_full_55550.sh: ligne 517: /bin/rm: No such file or directory
```

**Comment by pasqui23** [comment]
@rapenne-s Weird, It did kept the credentials for GOG.

In fact I would consider it ready, the problems with steam were a mistake in my `configuration.nix`

**Review by AndersonTorres — CHANGES_REQUESTED** [review]
Squash all commits into one.

File: pkgs/games/gamehub/default.nix [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+  nativeBuildInputs = [
+    desktop-file-utils
+    meson
+    ninja
+    pkg-config
+    vala
+    wrapGAppsHook
+  ];
+
+  propagatedBuildInputs = [
```
Comment (SuperSandro2000) [inline pkgs/games/gamehub/default.nix:42]: Why are all the dependencies propagated?

**Comment by MalteT** [comment]
Result of `nixpkgs-review pr 158044` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>gamehub</li>
  </ul>
</details>

**Comment by MalteT** [comment]
Here's the build log, might be something on my part, I'm not sure: https://gist.github.com/MalteT/30adc398eaefb603e98a0b7a88720ecc

**Comment by pasqui23** [comment]
Weird, on my machine it builds???

**Comment by rapenne-s** [comment]
Result of `nixpkgs-review pr 158044` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>gamehub</li>
  </ul>
</details>

**Comment by pasqui23** [comment]
K why `nix build .#gamehub` succeeds while `nixpkgs-review` fails?

**Comment by rapenne-s** [comment]
> K why `nix build .#gamehub` succeeds while `nixpkgs-review` fails?

do you have the exact same branch as the one available on GitHub? No local change? I'll try without `nixpkgs-review` just to be sure

**Comment by rapenne-s** [comment]
There is a 0.16.3-2-master release for gamehub on https://github.com/tkashkin/GameHub/releases which says it fixes building with newer Vala.

I figured out why `nixpkgs-review` was behaving differently. By default, it rebases the branch on top of `master`, so it's compiling using latest version of all build inputs. Newer vala got merged in between. More information at https://github.com/Mic92/nixpkgs-review#checkout-strategy-recommend-for-r-ryantm--cachix

Using the command `nixpkgs-review pr --checkout commit 158044` to build gamehub using the inputs from its branch version gives a success.

**Comment by MalteT** [comment]
Can confirm, builds after `gh pr checkout 158044`. `nixpkgs-review pr --checkout commit 158044` works aswell, so this should be fine then?

**Comment by SuperSandro2000** [comment]
@ofborg build gamehub

**Comment by SuperSandro2000** [comment]
> There must be an issue with `nixpkgs-review` here.

No, the reason behind it is that both nixpkgs-review and ofborg merge the PR into master before and there the build is broken.

**Comment by rapenne-s** [comment]
> > There must be an issue with `nixpkgs-review` here.
> 
> No, the reason behind it is that both nixpkgs-review and ofborg merge the PR into master before and there the build is broken.

this is what I figured out later in the thread. There is also a new gamehub release that's supposed to fix the build.

**Comment by pasqui23** [comment]
Ok updated it to the latest. lat's merge this before and not `nixpkgs-review pr` succeeds.

Let's merge this before it breaks again

**Comment by rapenne-s** [comment]
Result of `nixpkgs-review pr 158044` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>gamehub</li>
  </ul>
</details>

**Review by rapenne-s — APPROVED** [review]

**Comment by rapenne-s** [comment]
@pasqui23 how do you log into only one service (like GOG) and then use gamehub? When I log in with GOG and click on "ignore" button on the top right corner, I can see my GOG library, but when I restart gamehub I need to login again. Unsure if it's my doing wrong or a bug in gamehub

**Comment by MalteT** [comment]
@rapenne-s it seems to work for me, I just have to click 'Skip' after every start. But I'm not logged out when restarting gamehub

File: pkgs/games/gamehub/default.nix [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+    libmanette
+    libsoup
+    libXtst
+    sqlite
+    webkitgtk
+  ];
+
+  meta = with lib; {
+    homepage = "https://tkashkin.github.io/projects/gamehub";
+    description = "All your games in one place";
```
Comment (AndersonTorres) [inline pkgs/games/gamehub/default.nix:57]:
```suggestion
    description = "Unified library for all your games";
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Review by AndersonTorres — APPROVED** [review]
Looks fine to me.

**Comment by SuperSandro2000** [comment]
@ofborg build gamehub

**Pull request merged**